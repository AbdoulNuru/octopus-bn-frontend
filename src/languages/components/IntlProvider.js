import React, { Component } from 'react';
import { IntlProvider } from 'react-intl';
import { connect } from 'react-redux';

class Provider extends Component {
	constructor(props) {
		super(props);
	}

	render() {
		return (
			<IntlProvider locale={this.props.languageState.locale} messages={this.props.languageState.messages}>
				{this.props.children}
			</IntlProvider>
		);
	}
}

const mapStateToProps = (state) => {
	return {
		languageState: state.language
	};
};

const mapDispatchToProps = (dispatch) => {
	return {
		setLanguage: (language) => dispatch(loginAction(language))
	};
};

export default connect(mapStateToProps, mapDispatchToProps)(Provider);
