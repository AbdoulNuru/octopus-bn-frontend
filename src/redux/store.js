import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import rootReducer from './reducers';
import logger from 'redux-logger';
const middlwares = applyMiddleware(thunk, promise, logger);
const store = createStore(rootReducer, {}, composeWithDevTools(middlwares));

export default store;
