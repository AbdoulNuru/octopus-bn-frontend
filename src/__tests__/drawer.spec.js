import ResponsiveDrawer from "../components/ResponsiveDrawer";
import React from "react";
import { render, fireEvent, cleanup, waitFor } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../redux/store";
import userEvent from "@testing-library/user-event";
import IntlProvider from "../languages/components/IntlProvider";
import { BrowserRouter as Router, Link } from "react-router-dom";

const LostComponent = () => {
  return render(
    <Provider store={store}>
      <IntlProvider>
        <Router>
          <ResponsiveDrawer />
        </Router>
      </IntlProvider>
    </Provider>
  );
};

describe("LOST", () => {
  afterEach(cleanup);
  it("should render ResponsiveDrawer component", () => {
    localStorage.setItem(
      "bn-user-data",
      '{"id":"d01cf3f2-4601-4b53-8ffd-fd46b6ded623","method":"local","firstName":"Izabayo","lastName":"Blaise","email":"octopusbn@gmail.com","isVerified":true,"isUpdated":true,"gender":"male","birthDate":"1998-02-20T00:00:00.000Z","preferedLang":"en","preferedCurrency":"USD","residence":"kimironko","department":"travel","managerEmail":"needs.grid@gmail.com","imageUrl":"https://www.google.com/url?sa=i&url=https%3A%2F%2Fwww.freepik.com%2Ffree-photos-vectors%2Fman-profile&psig=AOvVaw1-_OdOwQ-SYcfMGvGLOYb4&ust=1582456400155000&source=images&cd=vfe&ved=0CAIQjRxqFwoTCOjsvNCD5ecCFQAAAAAdAAAAABAD","bio":"I love travel administrator","passportNumber":"RW1234567","role":"travel_administrator","notifyByEmail":true,"createdAt":"2020-05-05T13:11:19.884Z","updatedAt":"2020-05-05T13:11:19.884Z"}'
    );
    const { asFragment } = LostComponent();
    expect(asFragment(<ResponsiveDrawer />)).toMatchSnapshot();
  });
});
