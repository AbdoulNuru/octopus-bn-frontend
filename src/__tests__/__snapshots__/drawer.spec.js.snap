// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LOST should render ResponsiveDrawer component 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-4 mui-fixed MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="open drawer"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-5 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="makeStyles-search-9"
        >
          <div
            class="makeStyles-searchIcon-10"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <div
            class="MuiInputBase-root makeStyles-inputRoot-11"
          >
            <input
              aria-label="search"
              class="MuiInputBase-input makeStyles-inputInput-12"
              placeholder="Searchâ€¦"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="makeStyles-grow-2"
        />
        <div>
          <button
            aria-label="show 17 new notifications"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <span
                class="MuiBadge-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
                  />
                </svg>
                <span
                  class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary"
                >
                  17
                </span>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-controls="primary-search-account-menu"
            aria-haspopup="true"
            aria-label="account of current user"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeEnd"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circle makeStyles-orange-14"
              >
                <img
                  class="MuiAvatar-img"
                  src="/static/images/avatar/1.jpg"
                />
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </header>
    <nav
      aria-label="mailbox folders"
      class="makeStyles-drawer-3"
    >
      <div
        class="PrivateHiddenCss-smUp-123"
      />
      <div
        class="PrivateHiddenCss-xsDown-121"
      >
        <div
          class="MuiDrawer-root MuiDrawer-docked"
        >
          <div
            class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-7 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
          >
            <div>
              <div
                class="makeStyles-toolbar-151"
              />
              <hr
                class="MuiDivider-root"
              />
              <ul
                class="MuiList-root MuiList-padding"
              >
                <a
                  href="/dashboard"
                  style="text-decoration: none; color: rgb(0, 0, 0);"
                >
                  <div
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiListItemIcon-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                        />
                      </svg>
                    </div>
                    <div
                      class="MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                      >
                        Dashboard
                      </span>
                    </div>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </div>
                </a>
                <a
                  href="/requests"
                  style="text-decoration: none; color: rgb(0, 0, 0);"
                >
                  <div
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiListItemIcon-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M2.5 19h19v2h-19v-2zm19.57-9.36c-.21-.8-1.04-1.28-1.84-1.06L14.92 10l-6.9-6.43-1.93.51 4.14 7.17-4.97 1.33-1.97-1.54-1.45.39 2.59 4.49s7.12-1.9 16.57-4.43c.81-.23 1.28-1.05 1.07-1.85z"
                        />
                      </svg>
                    </div>
                    <div
                      class="MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                      >
                        My Requests
                      </span>
                    </div>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </div>
                </a>
                <a
                  href="/user-roles"
                  style="text-decoration: none; color: rgb(0, 0, 0);"
                >
                  <div
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiListItemIcon-root"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        style="width: 25px;"
                        viewBox="0 0 640 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M610.5 373.3c2.6-14.1 2.6-28.5 0-42.6l25.8-14.9c3-1.7 4.3-5.2 3.3-8.5-6.7-21.6-18.2-41.2-33.2-57.4-2.3-2.5-6-3.1-9-1.4l-25.8 14.9c-10.9-9.3-23.4-16.5-36.9-21.3v-29.8c0-3.4-2.4-6.4-5.7-7.1-22.3-5-45-4.8-66.2 0-3.3.7-5.7 3.7-5.7 7.1v29.8c-13.5 4.8-26 12-36.9 21.3l-25.8-14.9c-2.9-1.7-6.7-1.1-9 1.4-15 16.2-26.5 35.8-33.2 57.4-1 3.3.4 6.8 3.3 8.5l25.8 14.9c-2.6 14.1-2.6 28.5 0 42.6l-25.8 14.9c-3 1.7-4.3 5.2-3.3 8.5 6.7 21.6 18.2 41.1 33.2 57.4 2.3 2.5 6 3.1 9 1.4l25.8-14.9c10.9 9.3 23.4 16.5 36.9 21.3v29.8c0 3.4 2.4 6.4 5.7 7.1 22.3 5 45 4.8 66.2 0 3.3-.7 5.7-3.7 5.7-7.1v-29.8c13.5-4.8 26-12 36.9-21.3l25.8 14.9c2.9 1.7 6.7 1.1 9-1.4 15-16.2 26.5-35.8 33.2-57.4 1-3.3-.4-6.8-3.3-8.5l-25.8-14.9zM496 400.5c-26.8 0-48.5-21.8-48.5-48.5s21.8-48.5 48.5-48.5 48.5 21.8 48.5 48.5-21.7 48.5-48.5 48.5zM224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm201.2 226.5c-2.3-1.2-4.6-2.6-6.8-3.9l-7.9 4.6c-6 3.4-12.8 5.3-19.6 5.3-10.9 0-21.4-4.6-28.9-12.6-18.3-19.8-32.3-43.9-40.2-69.6-5.5-17.7 1.9-36.4 17.9-45.7l7.9-4.6c-.1-2.6-.1-5.2 0-7.8l-7.9-4.6c-16-9.2-23.4-28-17.9-45.7.9-2.9 2.2-5.8 3.2-8.7-3.8-.3-7.5-1.2-11.4-1.2h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c10.1 0 19.5-3.2 27.2-8.5-1.2-3.8-2-7.7-2-11.8v-9.2z"
                        />
                      </svg>
                    </div>
                    <div
                      class="MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                      >
                        Set role
                      </span>
                    </div>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </div>
                </a>
                <a
                  href="/profile-settings"
                  style="text-decoration: none; color: rgb(0, 0, 0);"
                >
                  <div
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiListItemIcon-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                        />
                      </svg>
                    </div>
                    <div
                      class="MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                      >
                        My Profile
                      </span>
                    </div>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </div>
                </a>
                <a
                  href="/add-accommodation"
                  style="text-decoration: none; color: rgb(0, 0, 0);"
                >
                  <div
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiListItemIcon-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6zm16-4H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-1 9h-4v4h-2v-4H9V9h4V5h2v4h4v2z"
                        />
                      </svg>
                    </div>
                    <div
                      class="MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                      >
                        Add accommodation
                      </span>
                    </div>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </div>
                </a>
                <a
                  href="/add-rooms"
                  style="text-decoration: none; color: rgb(0, 0, 0);"
                >
                  <div
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiListItemIcon-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M17 19.22H5V7h7V5H5c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2v-7h-2v7.22z"
                        />
                        <path
                          d="M19 2h-2v3h-3c.01.01 0 2 0 2h3v2.99c.01.01 2 0 2 0V7h3V5h-3V2zM7 9h8v2H7zM7 12v2h8v-2h-3zM7 15h8v2H7z"
                        />
                      </svg>
                    </div>
                    <div
                      class="MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                      >
                        Add rooms
                      </span>
                    </div>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </div>
                </a>
              </ul>
              <div
                aria-label="logoutClick"
                class="makeStyles-logout-158"
              >
                <hr
                  class="MuiDivider-root"
                />
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M13 3h-2v10h2V3zm4.83 2.17l-1.42 1.42C17.99 7.86 19 9.81 19 12c0 3.87-3.13 7-7 7s-7-3.13-7-7c0-2.19 1.01-4.14 2.58-5.42L6.17 5.17C4.23 6.82 3 9.26 3 12c0 4.97 4.03 9 9 9s9-4.03 9-9c0-2.74-1.23-5.18-3.17-6.83z"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      Logout
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </nav>
    <main
      class="makeStyles-content-8"
    >
      <div
        class="makeStyles-toolbar-6"
      />
    </main>
  </div>
</DocumentFragment>
`;
